### PR TITLE
[crypto] bump rust version & sync proof systems

### PR DESCRIPTION
**Pull request by mimoo**
following https://github.com/o1-labs/proof-systems/pull/970

### CODE REVIEW

**Comment by mimoo** [comment]
!ci-build-me

**Comment by mimoo** [comment]
!ci-build-me

**Comment by mimoo** [comment]
!ci-build-me

**Comment by mimoo** [comment]
!ci-build-me

**Comment by mimoo** [comment]
!ci-build-me

**Review by mrmr1993 — APPROVED** [review]

**Review by joseandro — APPROVED** [review]

**Review by fabrizio-m — APPROVED** [review]

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Review by jspada — APPROVED** [review]
approved.

File: nix/rust.nix [hunk 1]
```diff
@@ -8,9 +8,9 @@ let
       # override stdenv.targetPlatform here, if neccesary
     };
   toolchainHashes = {
-    "1.63.0" = "sha256-KXx+ID0y4mg2B3LHp7IyaiMrdexF6octADnAtFIOjrY=";
+    "1.67.0" = "sha256-riZUc+R9V35c/9e8KJUE+8pzpXyl0lRXt3ZkKlxoY0g=";
     "nightly-2022-09-12" =
```
Comment (fabrizio-m) [inline nix/rust.nix:12]: I think this has to become "nightly-2023-02-05".

**Review by lk86 — APPROVED** [review]

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Pull request merged**